portfolio/src/main/webapp/comments.js

```diff
@@ -0,0 +1,153 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TODO: Rewrite with a general class to keep track of things more easily...
+
+/**
+ * Clear the message div
+ */
+function clearMessageDiv() {
+  document.getElementById("message-container").innerHTML = "";
+}
+
+
+/**
+ * Return the TOTAL number of undeleted comments currently in datastore 
+ */
+function numComments() {
+  (fetch('/num-comments').then(response => response.text()).then(num => {
+    return num;
+  }));
+}
+
+/**
+ * Very basic pagination feature...
+ * Needs to be refined to have an upper limit on number of page buttons
+ */
+function pagination() {
+  var numShown = getCommentLimit();
+  (fetch('/num-comments').then(response => response.text()).then(numComments =>
+  {
+    var numPages = Math.ceil(numComments/numShown);
+    var paginationDiv = document.getElementById("pagination");
+    for (var pageNumber = 0; pageNumber < numPages; pageNumber++) {
+      const thisPageNumber = pageNumber;
+
+      var pageElement = document.createElement("a");
+      pageElement.innerText = thisPageNumber;
+      pageElement.href = "#";
+      pageElement.addEventListener('click', () => {
+        //load comments on pageNumber
+        addMessage(thisPageNumber, getSortMethod);
+      });
+
+      paginationDiv.appendChild(pageElement);
+    }
+  }));
+}
+
+
+/**
+ * Using fetch request content from servlet and add to home page
+ */
+function addMessage(pageNumber = 0, sortMethod = "DESCENDING") {
+  /**
+   * GET does not support additional URL params in the body
+   * So...
+   * A novel hack around this (rather than rewrite the GET logic)
```

Comment:
Using a format string or StringBuilder-type function to build a query string for a GET request is fairly common and not a hack.

Reply:
oops!